be/src/olap/column_vector.cpp

```diff
@@ -0,0 +1,212 @@
+                local.reset(new ScalarColumnVectorBatch<CppTypeTraits<OLAP_FIELD_TYPE_DATETIME>::CppType>(type_info, is_nullable));
+                break;
+            case OLAP_FIELD_TYPE_CHAR:
+                local.reset(new ScalarColumnVectorBatch<CppTypeTraits<OLAP_FIELD_TYPE_CHAR>::CppType>(type_info, is_nullable));
+                break;
+            case OLAP_FIELD_TYPE_VARCHAR:
+                local.reset(new ScalarColumnVectorBatch<CppTypeTraits<OLAP_FIELD_TYPE_VARCHAR>::CppType>(type_info, is_nullable));
+                break;
+            case OLAP_FIELD_TYPE_HLL:
+                local.reset(new ScalarColumnVectorBatch<CppTypeTraits<OLAP_FIELD_TYPE_HLL>::CppType>(type_info, is_nullable));
+                break;
+            case OLAP_FIELD_TYPE_OBJECT:
+                local.reset(new ScalarColumnVectorBatch<CppTypeTraits<OLAP_FIELD_TYPE_OBJECT>::CppType>(type_info, is_nullable));
+                break;
+            default:
+                return Status::NotSupported("unsupported type for ColumnVectorBatch: " + std::to_string(type_info->type()));
+        }
+        RETURN_IF_ERROR(local->resize(init_capacity));
+        *column_vector_batch = std::move(local);
+        return Status::OK();
+    } else {
+        switch (type_info->type()) {
+        case FieldType::OLAP_FIELD_TYPE_ARRAY: {
+            if (field == nullptr) {
+                return Status::NotSupported("When create ArrayColumnVectorBatch, `Field` is indispensable");
+            }
+            std::unique_ptr<ColumnVectorBatch> local(new ArrayColumnVectorBatch(type_info, is_nullable, init_capacity, field));
+            RETURN_IF_ERROR(local->resize(init_capacity));
+            *column_vector_batch = std::move(local);
+            return Status::OK();
+        }
+        default:
+            return Status::NotSupported("unsupported type for ColumnVectorBatch: " + std::to_string(type_info->type()));
+        }
+    }
+}
+
+template <class ScalarType>
+ScalarColumnVectorBatch<ScalarType>::ScalarColumnVectorBatch(const TypeInfo* type_info, bool is_nullable)
+: ColumnVectorBatch(type_info, is_nullable), _data(0) {
+}
+
+template <class ScalarType>
+ScalarColumnVectorBatch<ScalarType>::~ScalarColumnVectorBatch() = default;
+
+template <class ScalarType>
+Status ScalarColumnVectorBatch<ScalarType>::resize(size_t new_cap) {
+    if (capacity() < new_cap) { // before first init, _capacity is 0.
+        RETURN_IF_ERROR(ColumnVectorBatch::resize(new_cap));
+        _data.resize(new_cap);
+    }
+    return Status::OK();
+}
+
+ArrayColumnVectorBatch::ArrayColumnVectorBatch(
+        const TypeInfo* type_info,
+        bool is_nullable,
+        size_t init_capacity,
+        Field* field) : ColumnVectorBatch(type_info, is_nullable), _data(0), _item_offsets(1) {
+    auto array_type_info = reinterpret_cast<const ArrayTypeInfo*>(type_info);
+    _item_offsets[0] = 0;
+    ColumnVectorBatch::create(
+            init_capacity * 2,
```

Comment:
why double init_capacity here

Reply:
This is based on item's number is usually bigger than array's.